Packs/ApiModules/Scripts/OktaApiModule/OktaApiModule.py

```diff
@@ -0,0 +1,246 @@
+    RS256 = 'RS256'
+    RS384 = 'RS384'
+    RS512 = 'RS512'
+    ES256 = 'ES256'
+    ES384 = 'ES384'
+    ES512 = 'ES512'
+
+
+class AuthType(Enum):
+    API_TOKEN = 1
+    OAUTH = 2
+    NO_AUTH = 3
+
+
+class OktaClient(BaseClient):
+    def __init__(self, api_token: str, auth_type: AuthType = AuthType.API_TOKEN,
+                 client_id: str | None = None, scopes: list[str] | None = None, private_key: str | None = None,
+                 jwt_algorithm: JWTAlgorithm | None = None, *args, **kwargs):
+        """
+        Args:
+            api_token (str): API token for authentication.
+            auth_type (AuthType, optional): The type of authentication to use.
+            client_id (str | None, optional): Client ID for OAuth authentication (required if 'auth_type' is AuthType.OAUTH).
+            scopes (list[str] | None, optional): A list of scopes to request for the token
+                (required if 'auth_type' is AuthType.OAUTH).
+            private_key (str | None, optional): Private key for OAuth authentication (required if 'auth_type' is AuthType.OAUTH).
+            jwt_algorithm (str | None, optional): The algorithm to use for JWT signing
+                (required if 'auth_type' is AuthType.OAUTH).
+        """
+        super().__init__(*args, **kwargs)
+        self.api_token = api_token
+        self.auth_type = auth_type
+
+        missing_required_params = []
+
+        if self.auth_type == AuthType.OAUTH:
+            if not client_id:
+                missing_required_params.append('Client ID')
+
+            if not scopes:
+                missing_required_params.append('Scopes')
+
+            if not jwt_algorithm:
+                missing_required_params.append('JWT algorithm')
+
+            if not private_key:
+                missing_required_params.append('Private key')
+
+            if missing_required_params:
+                raise ValueError(f'Required OAuth parameters are missing: {", ".join(missing_required_params)}')
+
+            # Set type of variables non-optional after we assured they're assigned for mypy
+            self.client_id: str = client_id  # type: ignore
+            self.scopes: list[str] = scopes  # type: ignore
+            self.private_key: str = private_key  # type: ignore
+            self.jwt_algorithm: JWTAlgorithm = jwt_algorithm  # type: ignore
+
+        self.initial_setup()
+
+    def assign_app_role(self, client_id: str, role: str, auth_type: AuthType = AuthType.API_TOKEN) -> dict:
+        """
+        Assign a role to a client application.
+
+        Args:
+            client_id (str): The ID of the client application.
+            role (str): The role to assign to the client application.
+            auth_type (AuthType, optional): Authentication type to use for the request. Defaults to AuthType.API_TOKEN.
+
+        Returns:
+            dict: The response from the API.
+        """
+        return self._http_request(
+            auth_type=auth_type,
```

Comment:
As you use _http_request (from the baseClient) `auth_type` is not defined as expeted arg, so it's confusing
please remove if not relevant
remove it also from the method definition